testsupport/tiers/checks.go

```diff
@@ -37,6 +37,7 @@ const (
 	baseextendedidling = "baseextendedidling"
 	baselarge          = "baselarge"
 	testTier           = "test"
+	featuredTier       = "ftier" // dynamic tier created by nstemplatetier test to test feature toggles. It's a copy of the "base1ns" tier with additional resources managed by feature toggles. And since it's a copy of another tier we need to keep the name short to avoid exceeding 63 characters for the tier template names
```

Comment:
Originally I used the custom tier when provisioned a user:
```
NewSignupRequest(...).SpaceTier(customTier)
```
And it required defining that custom dynamic tier in checks.go. But it turned out not working properly anyway, so I switched to provisioning a space for a built-in tier and then promote it to the custom tier. This dynymic tier definition is a leftover. Let me clean it up now.

Reply:
https://github.com/codeready-toolchain/toolchain-e2e/pull/1008/commits/57aeb2d94e18c4c20124931a96643bae83050448

Reply:
OK. I faced the same issue with `VerifyResourcesProvisionedForSpaceWithCustomTier()`. The problem is that exiting tests do not really modify the templates themselves. They create copies of existing tier templates as is.

Modifying the templates after they are already duplicated is not working because such templates (duplicates) already exist: https://github.com/codeready-toolchain/toolchain-e2e/blob/master/testsupport/tiers/tier_setup.go#L143

It is still possible to modify an existing template by creating a new one (using the custom template as the original one, so we would duplicate a template of the existing custom template). However in this case the template.Spec.Tier would refer to the custom tier, so checks.go can't find the check for it.

This change should fix the problem: https://github.com/codeready-toolchain/toolchain-e2e/pull/1008/commits/ddc4d6a1b2907ca085f6d5685499330bcc078b27
It basically allows to provide the template modifiers during the custom tier creation.